src/components/radio/radio.ts

```diff
@@ -69,7 +69,10 @@ export class MdRadioGroup implements AfterContentInit, ControlValueAccessor {
   private _value: any = null;
 
   /** The HTML name attribute applied to radio buttons in this group. */
-  private _name: string = null;
+  private _name: string = `md-radio-group-${_uniqueIdCounter++}`;
+
+  /** @internal */
+  nameEmitter: EventEmitter<string> = new EventEmitter<string>();
```

Comment:
That could work - that was kind of what I was suggesting when we spoke earlier when I wasn't sure if I liked the approach I took here.